kanidmd/src/lib/idm/server.rs

```diff
@@ -1356,6 +1359,46 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
             })
     }
 
+    /// Generate a new set of backup code and remove the old ones.
+    pub fn generate_backup_code(
+        &mut self,
+        au: &mut AuditScope,
+        gbe: &GenerateBackupCodeEvent,
+    ) -> Result<String, OperationError> {
```

Comment:
Shouldn't we return a Vec<String> here to the client, and it's up to the front end utility to format this? Given this is structured data after all, and we don't want the client to have to str split on, when they can just process the json which is structured.

Reply:
Also consider adding a test into `kanidm_client/tests/proto_v1_test.rs` which actually tests these api interactions to create and retrieve the codes, then to use them from an external client perspective.